streamflow/core/scheduling.py

```diff
@@ -2,7 +2,7 @@
 
 import os
 from abc import ABC, abstractmethod
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Type, cast
```

Comment:
This goes into the `TYPE_CHECKING` section

Reply:
Keeping `Type` and `cast` inside the `TYPE_CHECKING` section throws a `NameError`. In particular, in the  `load` method of `HardwareRequirement`. This case was never tested, so in the `test_cwl_schedule_step` function of the `test_cwl_persistence` was added the `HardwareRequirement`. This new test fails leaving `Type` and `cast` inside the `TYPE_CHECKING` section.